msupply-foundation/open-msupply
# 3338 fix error popup when creating inbound return from outbound shipment

Fixes #3338

# 👩🏻‍💻 What does this PR do? 
 

Before changes: creating inbound return from outbound shipment showed a `Could not get inbound return` error:
![image](https://github.com/msupply-foundation/open-msupply/assets/55115239/3010414e-0fa6-4bf5-8c27-52c417e8d14c)


Now we don't see an error 🎉 
![Screenshot 2024-03-22 at 9 05 15 AM](https://github.com/msupply-foundation/open-msupply/assets/55115239/613a7a9c-39dc-429a-83c1-3f4ca5bd2bd0)

We were getting this error because the AddBatch button checks whether or not the inbound return is disabled. The add batch button isn't even visible in this view, but the hook was still being called (on a return that doesn't exist), hence the error!

Refactored the `AddBatch` button to it's own component, so that the hook will only be called when the button is visible, and the return exists.

# 🧪 How has/should this change been tested? 


- Go to an outbound shipment that is shipped or later
- Select some lines to return
- Modal opens, you should see no error
